src/Sessions.cc

```diff
@@ -81,11 +81,14 @@ void NetSessions::NextPacket(double t, Packet* pkt)
 	packet_mgr->ProcessPacket(pkt);
 	}
 
-void NetSessions::DoNextPacket(double t, const Packet* pkt)
+void NetSessions::DoNextPacket(double t, const Packet* pkt, size_t remaining)
 	{
 	const std::unique_ptr<IP_Hdr>& ip_hdr = pkt->ip_hdr;
 
-	uint32_t caplen = pkt->cap_len - pkt->hdr_size;
+	// Caplen is the amount of bytes left in the packet after all of the
+	// headers are removed.
+	uint32_t caplen = remaining;
```

Comment:
Wouldn't it make sense to remove `caplen` and just use `remaining` in the following? I also associate `caplen` with captured length with might differ from actual packet length. To me semantics seem mixed up here.